src/loaders/loaderUtils.ts

```diff
@@ -157,3 +158,99 @@ export async function fetchSubjects(schemaRegistry: SchemaRegistry): Promise<str
 
   return (await client.list()).sort();
 }
+
+/**
+ * Given a schema registry and a subject, fetch the versions available, then fetch the details
+ * of each version and return them as an array of {@link Schema}.
+ *
+ * The returned array of schema metadata concerning a a single subject is called a "subject group".
+ */
+export async function fetchSchemaSubjectGroup(
+  schemaRegistry: SchemaRegistry,
+  subject: string,
+): Promise<Schema[]> {
+  const sidecarHandle = await getSidecar();
+  const client: SubjectsV1Api = sidecarHandle.getSubjectsV1Api(
+    schemaRegistry.id,
+    schemaRegistry.connectionId,
+  );
+
+  // Learn all of the live version numbers for the subject in one round trip.
+  const versions: number[] = await client.listVersions({ subject });
+
+  // Now prep to fetch each of the versions concurrently via concurrent
+  // calls to fetchSchemaVersion() driven by executeInWorkerPool().
+  const highestVersion = Math.max(...versions);
+  const concurrentVersionRequests: FetchSchemaVersionParams[] = versions.map(
+    (version): FetchSchemaVersionParams => {
+      return {
+        // The only varying parameter in the concurrent calls/requests is the version number.
+        schemaRegistry: schemaRegistry,
+        client: client,
+        subject: subject,
+        version: version,
+        highestVersion: highestVersion,
+      };
+    },
+  );
+
+  // Fetch all versions concurrently capped at 5 concurrent requests at a time.
+  const concurrentFetchResults = await executeInWorkerPool(
+    fetchSchemaVersion,
+    concurrentVersionRequests,
+    {
+      maxWorkers: 5,
+    },
+  );
+
+  // Filter the executeInWorkerPool() return for successful results and return the schemas.
+  // If any single request failed, fail the whole operation.
+  const schemas: Schema[] = new Array(concurrentFetchResults.length);
+  concurrentFetchResults.forEach((result, index) => {
+    if (isSuccessResult(result)) {
+      schemas[index] = result.result;
+    } else {
+      // improve this before PR raised.
+      throw result.error;
```

Comment:
👀

Reply:
Not going to know how to handle better at this time. Revised comment spelling for now.